app/controllers/queue_items_controller.rb

```diff
@@ -0,0 +1,36 @@
+class QueueItemsController < ApplicationController
+
+  before_filter :require_logged_in
+
+  def index
+    if logged_in?
+      @queue_items = current_user.queue_items
+    else
+      redirect_to sign_in_path
+    end
+  end
+
+  def create
+
+    video = Video.find(params[:video_id])
+    if video && !current_user_queue_include?(video)
+      queue_item = QueueItem.new(user: current_user, video: video)
+      queue_item.position = current_user.queue_items.count + 1
+      queue_item.save
+    end
+    redirect_to my_queue_path
+  end
+
+  def destroy
+    queue_item = QueueItem.find(params[:id])
+    queue_item.destroy if queue_item.user == current_user
+    redirect_to my_queue_path
+  end
+
+private
+
+  def current_user_queue_include?(video)
```

Comment:
Actually, I think you do know how to do it. I just mean that you can create a method in your `User` model which provides this functionality. Like this:

```ruby
class User < ActiveRecord::Base
  def queue_includes?(video)
    queue_items.map(&:video).include?(video)
  end
end
```

Then you can use it here like this:

```ruby
current_user.queue_includes?(video)
```

A major advantage to this is that you would then have access to that method from everywhere in your app.